internal/framework/runnables/job.go

```diff
@@ -0,0 +1,52 @@
+package runnables
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+// JobConfig is the configuration for a job.
+type JobConfig struct {
+	// Worker is the function that will be run for every job iteration.
+	Worker func(context.Context)
+	// Logger is the logger.
+	Logger logr.Logger
+	// Period defines the period of the job. The job will run every Period.
+	Period time.Duration
+}
+
+// Job periodically runs a worker function.
+type Job struct {
```

Comment:
Should we call this `PeriodicJob` or `CronJob`?

Reply:
CronJob sounds good to me